scanners/flutter/flutterproject/pubspec.go

```diff
@@ -0,0 +1,72 @@
+package flutterproject
+
+import (
+	"io"
+	"path/filepath"
+
+	"gopkg.in/yaml.v3"
+)
+
+const pubspecRelPath = "pubspec.yaml"
+
+type PubspecVersionReader struct {
+	fileOpener FileOpener
+}
+
+func NewPubspecVersionReader(fileOpener FileOpener) PubspecVersionReader {
+	return PubspecVersionReader{
+		fileOpener: fileOpener,
+	}
+}
+
+func (r PubspecVersionReader) ReadSDKVersions(projectRootDir string) (*VersionConstraint, *VersionConstraint, error) {
+	pubspecPth := filepath.Join(projectRootDir, pubspecRelPath)
+	f, err := r.fileOpener.OpenFile(pubspecPth)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if f == nil {
+		return nil, nil, nil
+	}
+
+	flutterVersionStr, dartVersionStr, err := parsePubspecSDKVersions(f)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var flutterVersion *VersionConstraint
+	if flutterVersionStr != "" {
+		flutterVersion, err = NewVersionConstraint(flutterVersionStr, PubspecVersionSource)
+		if err != nil {
+			return nil, nil, err
+		}
+	}
+
+	var dartVersion *VersionConstraint
+	if dartVersionStr != "" {
+		dartVersion, err = NewVersionConstraint(dartVersionStr, PubspecVersionSource)
+		if err != nil {
+			return nil, nil, err
```

Comment:
Not every Flutter project will have the Dart SDK version specified in the pubspec and I think even nowadays the Flutter version is not required.

Is this really an error scenario?

See our demo project: https://github.com/bitrise-io/flutter-demo-app/blob/main/pubspec.yaml#L21

Reply:
Ah I missed the part where we're checking if the string is not empty first 🤦‍♂️